### PR TITLE
[debug-certificate-manager] Option to skip automatically trusting and untrusting a certificate

### PR DESCRIPTION
**Pull request by apostolisms**
## Summary
Adds the ability to pass a skipCertificateTrust option when calling ensureCertificateAsync in order to skip automatically trusting and untrusting generated certificates

## Details
Adds the ability to pass a skipCertificateTrust option when calling ensureCertificateAsync in order to skip automatically trusting and untrusting generated certificates. This is to handle scenarios that execute with sudo and require a password input but there is no terminal available. i.e. VSCode extension.

## How it was tested
manually tested:

- no certs present on disk
- valid certs present on disk
- invalid certs present on disk

## Impacted documentation

### CODE REVIEW

File: common/changes/@rushstack/debug-certificate-manager/user-apc-skipcerttrust_2024-09-20-22-22.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "@rushstack/debug-certificate-manager",
+      "comment": "Option to skip automatically trusting and untrusting a certificate",
```
Comment (iclanton) [inline common/changes/@rushstack/debug-certificate-manager/user-apc-skipcerttrust_2024-09-20-22-22.json:5]:
```suggestion
      "comment": "Add a `skipCertificateTrust` option to `CertificateManager.ensureCertificateAsync` that skips automatically trusting the generated certificate and untrusting an existing certificate with issues.",
```

File: common/reviews/api/debug-certificate-manager.api.md [hunk 2]
```diff
@@ -39,6 +39,8 @@ export interface ICertificate {
 
 // @public
 export interface ICertificateGenerationOptions {
+    // (undocumented)
```
Comment (iclanton) [inline common/reviews/api/debug-certificate-manager.api.md:42]: You'll probably need to rerun the build and update this.

**Review by iclanton — APPROVED** [review]

File: libraries/debug-certificate-manager/src/CertificateManager.ts [hunk 3]
```diff
@@ -110,6 +110,10 @@ export interface ICertificateGenerationOptions {
    * How many days the certificate should be valid for.
    */
   validityInDays?: number;
+  /*
```
Comment (iclanton) [inline libraries/debug-certificate-manager/src/CertificateManager.ts:113]:
```suggestion
  /**
```

File: common/reviews/api/debug-certificate-manager.api.md (hunk 2 above)
Comment (iclanton) [reply common/reviews/api/debug-certificate-manager.api.md:42]:
```suggestion
```

**Review by dmichon-msft — APPROVED** [review]

**Pull request merged**